tests/bugs/bool-op.slang

```diff
@@ -0,0 +1,77 @@
+// enum.slang
+//TEST(compute):COMPARE_COMPUTE_EX:-slang -compute
+//TEST(compute, vulkan):COMPARE_COMPUTE_EX:-vk -compute
+
+// Confirm operations that produce bools - such as comparisons, or && ||, ! work correctly
+
+//TEST_INPUT:ubuffer(data=[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0], stride=4):dxbinding(0),glbinding(0),out
+RWStructuredBuffer<int> outputBuffer;
+
+[numthreads(16, 1, 1)]
+void computeMain(uint3 dispatchThreadID : SV_DispatchThreadID)
+{
+    uint tid = dispatchThreadID.x;; 
+
+    uint uv = tid;
+    int iv = int(tid);
+    
+    bool2 bv2 = { tid & 1, tid > 10 };      
+    
+    float2 f2 = { float(tid), float(tid + 1) };
+    
+    bool bv = tid > 6;
+    let not_bv2 = !bv2;
+    
+    int r = 0;
+    if (bv)
+    {
+        r |= 0x0001;
+    }
+    if (!bv)
+    {
+        r |= 0x0002;
+    }
+    
+    if (iv)
+    {
+        r|= 0x0004;
+    }
+    if (!iv)
+    {
+        r|= 0x0008;
+    } 
+    
+    if (uv)
+    {
+        r |= 0x0010;
+    }
+    if (!uv)
+    {
+        r |= 0x0020;
+    }
+    
+    if (all(bv2))
+    {
+        r |= 0x0040;
+    }
+    if (all(not_bv2))
+    {
+        r |= 0x0080;
+    }
+    
+    if (any(!f2))
+    {
+        r |= 0x0100;
+    }
+    
+    // TODO(JS): Support on GLSL targets
+    // This doesn't currently work on GLSL targets, and because there 
+    // do not appear to be any vector bitwise operations. Could be achieved
+    // by deconstructing, and reconstructing the vec result
+    /* if (all(f2 || bv2))
+    {
+        r |= 0x0200;
```

Comment:
To make sure I understand: GLSL doesn't support `||` or `&&` on vectors, because they have those operators short-circuit anyway. They also don't support `|` or `&` on boolean vectors, since they aren't arithmetic types.

We could either deconstruct and then reconstruct, or else do `left || right` as `bec2(int2(left) | int2(right))` to do the operation on integer vectors instead.

Either way, it seems like we need to eventually have a direct implementation of `||` and `&&` for vectors in the stdlib.

Reply:
Yes - I think perhaps the reason the bvec ops aren't there is you can case to ivec do the op and then cast back to bvec. So perhaps I'll add support for that, cos seems fairly simple.

Reply:
The one note I'd make is that on the long-term time scale I'd like to see which of these GLSL-specific problems we can/should handle in `emit.cpp` by emitting more complicated operations, and which we should handle in a preceding IR pass that just rewrites the missing operations out of the code. For now doing everything in `emit.cpp` is the easiest, but we should pay attention to whether we are making a mess of that file.